lib/tf-idf-similarity/bm25_model.rb

```diff
@@ -22,8 +22,12 @@ def inverse_document_frequency(term)
     #
     # @note Like Lucene, we use a b value of 0.75 and a k1 value of 1.2.
     def term_frequency(document, term)
-      tf = document.term_count(term)
-      (tf * 2.2) / (tf + 0.3 + 0.9 * documents.size / @model.average_document_size)
+      if @model.average_document_size.zero?
+        Float::NAN
```

Comment:
I think that this is not needed but I added this to keep backward compatibility.
Without this, some specs are failed:

```text
Failures:

  1) TfIdfSimilarity::BM25Model without documents #term_frequency should return the term frequency
     Failure/Error: model.tf(document, 'foo').should be_nan
       expected `0.0.nan?` to return true, got false
     # ./spec/bm25_model_spec.rb:79:in `block (4 levels) in <module:TfIdfSimilarity>'

  2) TfIdfSimilarity::BM25Model without documents #term_frequency_inverse_document_frequency should return negative infinity
     Failure/Error: model.tfidf(document, 'foo').should be_nan
       expected `0.0.nan?` to return true, got false
     # ./spec/bm25_model_spec.rb:85:in `block (4 levels) in <module:TfIdfSimilarity>'

Deprecation Warnings:

Using `should` from rspec-expectations' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` with `config.expect_with(:rspec) { |c| c.syntax = :should }` instead. Called from /tmp/tf-idf-similarity/spec/bm25_model_spec.rb:49:in `block (4 levels) in <module:TfIdfSimilarity>'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.

1 deprecation warning total

Finished in 0.0532 seconds (files took 0.16876 seconds to load)
102 examples, 2 failures, 3 pending

Failed examples:

rspec ./spec/bm25_model_spec.rb:78 # TfIdfSimilarity::BM25Model without documents #term_frequency should return the term frequency
rspec ./spec/bm25_model_spec.rb:84 # TfIdfSimilarity::BM25Model without documents #term_frequency_inverse_document_frequency should return negative infinity
```